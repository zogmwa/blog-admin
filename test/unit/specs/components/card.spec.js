import Vue from 'vue'
import ElementUI from 'element-ui'
import { mount } from 'vue-test-utils'

import Card from '@/components/card.vue'

Vue.use(ElementUI)

describe('App.vue', () => {
  let wrapper

  afterEach(() => {
    wrapper.destroy()
  })

  it('Has methods', () => {
    wrapper = mount(Card)
    expect(typeof wrapper.vm.toggle).toBe('function')
  })

  it('Has render props', () => {
    wrapper = mount(Card, {
      propsData: {
        width: '50px',
        type: [
          {
            name: '状态',
            typeName: 'state',
            list: [
              { name: '全部', id: '' },
              { name: '待审核', id: 0 },
              { name: '审核通过', id: 1 },
              { name: '审核不通过', id: 2 }
            ],
            default: ''
          }
        ],
        carLoading: true
      }
    })
    expect(wrapper.find('.text').hasStyle('width', '50px')).toBe(true)
    expect(wrapper.find('.el-loading-mask')).not.toBe(null)
    expect(wrapper.findAll('.el-radio-button--small').length).toBe(4)
    expect(wrapper.find('.el-radio-button--small').classes()).toContain('is-active')
  })

  it('Change event', () => {
    wrapper = mount(Card, {
      propsData: {
        width: '50px',
        type: [
          {
            name: '状态',
            typeName: 'state',
            list: [
              { name: '全部', id: '' },
              { name: '待审核', id: 0 },
              { name: '审核通过', id: 1 },
              { name: '审核不通过', id: 2 }
            ],
            default: ''
          }
        ],
        carLoading: false
      }
    })
    const radioGroup = wrapper.find('.el-radio-group')
    radioGroup.trigger('change')
    expect(wrapper.emitted().toggle).toBeTruthy()
  })

  it('Has the expected html structure', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})
