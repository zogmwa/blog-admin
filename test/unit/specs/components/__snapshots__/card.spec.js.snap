// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.vue Has the expected html structure 1`] = `
<div
  class="el-card box-card"
>
  <!---->
  <div
    class="el-card__body"
  >
    <div
      class="item"
    >
      <span
        class="text"
        style="width: 50px;"
      >
        状态：
      </span>
       
      <div
        class="el-radio-group"
        role="radiogroup"
      >
        <span
          name="list"
          tag="div"
        >
          <label
            aria-checked="true"
            class="el-radio-button btn el-radio-button--small is-active"
            role="radio"
            tabindex="0"
          >
            <input
              class="el-radio-button__orig-radio"
              tabindex="-1"
              type="radio"
            />
            <span
              class="el-radio-button__inner"
            >
              
          全部
        
              <!---->
            </span>
          </label>
          <label
            class="el-radio-button btn el-radio-button--small"
            role="radio"
            tabindex="-1"
          >
            <input
              class="el-radio-button__orig-radio"
              tabindex="-1"
              type="radio"
            />
            <span
              class="el-radio-button__inner"
            >
              
          待审核
        
              <!---->
            </span>
          </label>
          <label
            class="el-radio-button btn el-radio-button--small"
            role="radio"
            tabindex="-1"
          >
            <input
              class="el-radio-button__orig-radio"
              tabindex="-1"
              type="radio"
            />
            <span
              class="el-radio-button__inner"
            >
              
          审核通过
        
              <!---->
            </span>
          </label>
          <label
            class="el-radio-button btn el-radio-button--small"
            role="radio"
            tabindex="-1"
          >
            <input
              class="el-radio-button__orig-radio"
              tabindex="-1"
              type="radio"
            />
            <span
              class="el-radio-button__inner"
            >
              
          审核不通过
        
              <!---->
            </span>
          </label>
        </span>
      </div>
    </div>
     
  </div>
</div>
`;
